config/config/src/index.ts

```diff
@@ -41,138 +45,38 @@ type KebabCaseConfig = {
-  noproxy: String,
-  'npm-path': String,
-  offline: Boolean,
-  'only-built-dependencies': [String],
-  'pack-gzip-level': Number,
-  'package-import-method': ['auto', 'hardlink', 'clone', 'copy'],
-  'patches-dir': String,
-  pnpmfile: String,
-  'package-manager-strict': Boolean,
-  'package-manager-strict-version': Boolean,
-  'prefer-frozen-lockfile': Boolean,
-  'prefer-offline': Boolean,
-  'prefer-symlinked-executables': Boolean,
-  'prefer-workspace-packages': Boolean,
-  production: [null, true],
-  'public-hoist-pattern': Array,
-  'publish-branch': String,
-  'recursive-install': Boolean,
-  reporter: String,
-  'resolution-mode': ['highest', 'time-based', 'lowest-direct'],
-  'resolve-peers-from-workspace-root': Boolean,
-  'aggregate-output': Boolean,
-  'reporter-hide-prefix': Boolean,
-  'save-peer': Boolean,
-  'save-workspace-protocol': Boolean,
-  'script-shell': String,
-  'shamefully-flatten': Boolean,
-  'shamefully-hoist': Boolean,
-  'shared-workspace-lockfile': Boolean,
-  'shell-emulator': Boolean,
-  'side-effects-cache': Boolean,
-  'side-effects-cache-readonly': Boolean,
-  symlink: Boolean,
-  sort: Boolean,
-  'state-dir': String,
-  'store-dir': String,
-  stream: Boolean,
-  'strict-store-pkg-content-check': Boolean,
-  'strict-peer-dependencies': Boolean,
-  'use-beta-cli': Boolean,
-  'use-node-version': String,
-  'use-running-store-server': Boolean,
-  'use-store-server': Boolean,
-  'use-stderr': Boolean,
-  'verify-store-integrity': Boolean,
-  'virtual-store-dir': String,
-  'virtual-store-dir-max-length': Number,
-  'peers-suffix-max-length': Number,
-  'workspace-concurrency': Number,
-  'workspace-packages': [String, Array],
-  'workspace-root': Boolean,
-  'test-pattern': [String, Array],
-  'changed-files-ignore-pattern': [String, Array],
-  'embed-readme': Boolean,
-  'update-notifier': Boolean,
-  'registry-supports-time-field': Boolean,
-  'fail-if-no-match': Boolean,
-
-}, npmTypes.types)
-
 export type CliOptions = Record<string, unknown> & { dir?: string, json?: boolean }
 
-export async function getConfig (
-  opts: {
-    globalDirShouldAllowWrite?: boolean
-    cliOptions: CliOptions
-    packageManager: {
-      name: string
-      version: string
+export type IgnoreNonAuthSettingsFromLocal = false | { nonAuthConfigDir: string }
+
+export async function getConfig (opts: {
+  globalDirShouldAllowWrite?: boolean
+  cliOptions: CliOptions
+  packageManager: {
+    name: string
+    version: string
+  }
+  rcOptionsTypes?: Record<string, unknown>
+  workspaceDir?: string | undefined
+  checkUnknownSetting?: boolean
+  env?: Record<string, string | undefined>
+  ignoreNonAuthSettingsFromLocal?: IgnoreNonAuthSettingsFromLocal
+}): Promise<{ config: Config, warnings: string[] }> {
+  if (opts.ignoreNonAuthSettingsFromLocal) {
+    const { ignoreNonAuthSettingsFromLocal: { nonAuthConfigDir }, ...authOpts } = opts
+    const nonAuthOpts: typeof authOpts = {
```

Comment:
I am not sure about this variable name. Can't the global config have some non-overlapping auth settings too? I'd call this globalConfigOptsInstead.

Reply:
https://github.com/pnpm/pnpm/pull/8207/commits/a735be68b33179784c77c0d2b8bb5f13d814c996